### PR TITLE
Disambiguate shared and static libs `.lib` files on Windows using MSVC

### LINKED ISSUE #454: Error when building both static and shared libs on Windows with MSVC
Description
---

It's currently not possible to build both static and shared versions of libraries on Windows when using the MSVC-based toolchain because both variants produce a `.lib` file with the exact same output name.

For the shared libs it represents the export file and for static libs, the library itself. 
It would be nice to detect such cases and automatically suffix the output name of the static lib with `_static` to disambiguate. It is also well documented in a [`conda-forge` enhancement proposal](https://github.com/conda-forge/cfep/blob/main/cfep-18.md#windows).

I'll open a PR to discuss a possible fix!

Steps to reproduce
---

I'm building the project using a conda environment:
```
micromamba create -n sundials -y c-compiler cmake ninja
micromamba activate sundials
cmake -LAH -G "Ninja" -B build -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DBUILD_STATIC_LIBS=ON -S .
ninja -C build
```
Then you get at the last step:
```
ninja: error: build.ninja:997: multiple rules generate bin/sundials_core.lib
```

### PR DESCRIPTION
**Pull request by adriendelsalle**
Description
---

Disambiguate shared and static libs `.lib` files on Windows using MSVC
Suffix library output name with `_static` when building both static and shared libs on Windows

Closes #454

### CODE REVIEW

File: cmake/macros/SundialsAddLibrary.cmake [hunk 1]
```diff
@@ -325,10 +325,17 @@ macro(sundials_add_library target)
 
       # set the correct output name
       if(sundials_add_library_OUTPUT_NAME)
-        set_target_properties(${_actual_target_name} PROPERTIES
-          OUTPUT_NAME ${sundials_add_library_OUTPUT_NAME}
-          CLEAN_DIRECT_OUTPUT 1
-        )
+        if(WIN32 AND ${_libtype} MATCHES "STATIC")
```
Comment (mmuetzel) [inline cmake/macros/SundialsAddLibrary.cmake:328]: Consider changing this to the following:
``` suggestion
        if((MSVC OR ("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC")) AND ${_libtype} MATCHES "STATIC")
```
For MinGW, `WIN32` is true. But it doesn't have the naming conflict that compilers that target MSVC have.

Comment (adriendelsalle) [reply cmake/macros/SundialsAddLibrary.cmake:328]: Thanks for your help! I understood from @balos1 that this PR should be closed in favor to a more generic/customizable approach.
Let me know!

**Comment by balos1** [comment]
Please add a note about this change to `CHANGELOG.md` and `doc/shared/RecentChanges.rst`

**Comment by adriendelsalle** [comment]
@balos1 
- rebased on `develop`
- added changes to `CHANGELOG.md` and `doc/shared/RecentChanges.rst`
- fixed `SUNDIALS_EXPORT` macro incorrectly set when building both variants -> let me know if you prefer another PR for this, even if it's related to building both static and shared libs

**Review by balos1 — APPROVED** [review]

**Pull request merged**